client/state/form/reducer.js

```diff
@@ -21,8 +22,12 @@ reducers[ SET_FORM_PROPERTY ] = ( state, action ) => {
 	return Object.assign( {}, state, newObj );
 };
 
-reducers[ NEXT_STEP ] = ( state ) => {
-	return Object.assign( {}, state, { currentStep: state.currentStep + 1 } );
+reducers[ BACK_FROM_SUGGESTION ] = ( state ) => {
+	return [
+		setFormProperty( 'errors', {} ),
+		setFormProperty( 'acceptSuggestion', undefined ),
+		setFormProperty( 'pristine', false ),
+	].reduce( form, state );
```

Comment:
He uses an actions that encapsulates several actions. Which is more or less what we are doing with `redux-thunk`. But what I did here was to call several reducers **inside** a reducer, and that seems a little more weird.

About performance... I wouldn't worry about that, not at this level at least.